### PR TITLE
homeassistant: 0.111.4 -> 0.112.3

### PR DESCRIPTION
**Pull request by mweinelt**
Using the glorious `./update.sh` tooling. Thanks @Mic92!

###### Motivation for this change

New upstream release. 
https://www.home-assistant.io/blog/2020/07/01/release-112/

Logbook and History are now *fast*, expect db migrations on startup.

The tests needed a fixup, because the embedded MQTT broker was removed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@Mic92 @dotlambda

### CODE REVIEW

**Comment by mweinelt** [comment]
Upgrade on an RPi4 with PostgreSQL and 1.4GB database size on an SSD connected over USB3 took roughly 10 seconds.

```
2020-07-01 19:42:57 WARNING (Recorder) [homeassistant.components.recorder.migration] Database is about to upgrade. Schema version: 7
2020-07-01 19:42:57 WARNING (Recorder) [homeassistant.components.recorder.migration] Adding columns context_parent_id to table events. Note: this can take several minutes on large databases and slow computers. Please be patient!
2020-07-01 19:42:57 WARNING (Recorder) [homeassistant.components.recorder.migration] Adding columns old_state_id to table states. Note: this can take several minutes on large databases and slow computers. Please be patient!
2020-07-01 19:42:57 WARNING (Recorder) [homeassistant.components.recorder.migration] Adding index `ix_events_context_parent_id` to database. Note: this can take several minutes on large databases and slow computers. Please be patient!
2020-07-01 19:43:02 WARNING (Recorder) [homeassistant.components.recorder.migration] Adding index `ix_events_event_type_time_fired` to database. Note: this can take several minutes on large databases and slow computers. Please be patient!
2020-07-01 19:43:06 WARNING (MainThread) [homeassistant.setup] Setup of recorder is taking over 10 seconds.
2020-07-01 19:43:07 WARNING (Recorder) [homeassistant.components.recorder] Ended unfinished session (id=201 from 2020-06-30 05:30:37.667604+00:00)
```

The load times of logbook/history are down from ~20s to under less than 3s.

**Comment by colemickens** [comment]
This broke denonavr integration for me. The log has this, which is surprising since `defusedxml` is already in nixpkgs, so I assumed `update.sh` would've pulled it in. I only just started looking though (and I don't necessarily expect my issue to block this PR, just wanted to give initial feedback.)

**Comment by colemickens** [comment]
Well, as far as I can tell, denonavr->ssdp->defusedxml, so it looks like a possible bug in the update script and it's dependency resolution.

I'm not sure it matters to me anymore though. This release removes `denonavr`'s ability to be configured via YAML. I am looking for a more static home-assistant replacement moving forward. (Something nix-powered would be neat, and could pair nicely with esphome's declarative nature.)

File: pkgs/servers/home-assistant/component-packages.nix [hunk 1]
```diff
@@ -320,7 +320,7 @@
     "hangouts" = ps: with ps; [ ]; # missing inputs: hangups
     "harman_kardon_avr" = ps: with ps; [ ]; # missing inputs: hkavr
     "harmony" = ps: with ps; [ aioharmony];
-    "hassio" = ps: with ps; [ aiohttp-cors sqlalchemy]; # missing inputs: home-assistant-frontend
```
Comment (Mic92) [inline pkgs/servers/home-assistant/component-packages.nix:323]: Please generate this file using the latest `update.py`. This version is missing transitive dependencies.

**Comment by Mic92** [comment]
> Well, as far as I can tell, denonavr->ssdp->defusedxml, so it looks like a possible bug in the update script and it's dependency resolution.
> 
> I'm not sure it matters to me anymore though. This release removes `denonavr`'s ability to be configured via YAML. I am looking for a more static home-assistant replacement moving forward. (Something nix-powered would be neat, and could pair nicely with esphome's declarative nature.)

[mach-nix](https://discourse.nixos.org/t/mach-nix-create-python-environments-quick-and-easy/6858) could provide more precise pinning.

**Comment by Mic92** [comment]
> Well, as far as I can tell, denonavr->ssdp->defusedxml, so it looks like a possible bug in the update script and it's dependency resolution.
> 
> I'm not sure it matters to me anymore though. This release removes `denonavr`'s ability to be configured via YAML. I am looking for a more static home-assistant replacement moving forward. (Something nix-powered would be neat, and could pair nicely with esphome's declarative nature.)

Our latest script does resolve transitive dependencies. It was not used here for updating though.

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix:323]: This is in fact the latest version of parse-requirements.py:

```
commit d9068791d9be53a510fcd40f7379ace6fa4a12f7
Author: Jörg Thalheim <joerg@thalheim.io>
Date:   Sun Jun 21 16:07:30 2020 +0100

    home-assistant: also collect after_dependencies
```

Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix:323]: Resolved the dependency tree by hand:

- [hassio](https://github.com/home-assistant/core/blob/dev/homeassistant/components/hassio/manifest.json)
  - [http](https://github.com/home-assistant/core/blob/dev/homeassistant/components/http/manifest.json)
    - requires `aiohttp-cors`
  - [panel_custom](https://github.com/home-assistant/core/blob/dev/homeassistant/components/panel_custom/manifest.json)
    - [frontend](https://github.com/home-assistant/core/blob/dev/homeassistant/components/frontend/manifest.json)
      - [api](https://github.com/home-assistant/core/blob/dev/homeassistant/components/api/manifest.json)
        - http
      - [auth](https://github.com/home-assistant/core/blob/dev/homeassistant/components/auth/manifest.json)
        - http
      - [config](https://github.com/home-assistant/core/blob/dev/homeassistant/components/config/manifest.json)
        - http
      - [device_automation](https://github.com/home-assistant/core/blob/dev/homeassistant/components/device_automation/manifest.json)
      - http
      - [lovelace](https://github.com/home-assistant/core/blob/dev/homeassistant/components/lovelace/manifest.json)
      - [onboarding](https://github.com/home-assistant/core/blob/dev/homeassistant/components/onboarding/manifest.json)
        - auth
        - http
        - [person](https://github.com/home-assistant/core/blob/dev/homeassistant/components/person/manifest.json)
          - [device_tracker](https://github.com/home-assistant/core/blob/dev/homeassistant/components/device_tracker/manifest.json)
            - [zone](https://github.com/home-assistant/core/blob/dev/homeassistant/components/zone/manifest.json)
      - [search](https://github.com/home-assistant/core/blob/dev/homeassistant/components/search/manifest.json)
        - websocket_api
        - [scene](https://github.com/home-assistant/core/blob/dev/homeassistant/components/scene/manifest.json)
        - [group](https://github.com/home-assistant/core/blob/dev/homeassistant/components/group/manifest.json)
        - [automation](https://github.com/home-assistant/core/blob/dev/homeassistant/components/automation/manifest.json)
          - device_automation
          - [webhook](https://github.com/home-assistant/core/blob/dev/homeassistant/components/webhook/manifest.json)
            - http
        - [script](https://github.com/home-assistant/core/blob/dev/homeassistant/components/script/manifest.json)
      - [system_log](https://github.com/home-assistant/core/blob/dev/homeassistant/components/system_log/manifest.json)
        - http
      - [websocket_api](https://github.com/home-assistant/core/blob/dev/homeassistant/components/websocket_api/manifest.json)
        - http


The only requirement I can find in this tree is `aiohttp-cors`. In fact the only manifests that request `sqlalchemy` are that of the `recorder` and `sql` component.

**Comment by mweinelt** [comment]
> This broke denonavr integration for me. The log has this, which is surprising since `defusedxml` is already in nixpkgs, so I assumed `update.sh` would've pulled it in. I only just started looking though (and I don't necessarily expect my issue to block this PR, just wanted to give initial feedback.)
> 
> error: https://gist.github.com/colemickens/e814a6e78d29dc487336c242c9b59418

- denonavr
  - requires `"denonavr==0.9.3"` `"getmac==0.8.2"`

Does not depend on any other component, so the update script won't find defusedxml, which is a requirement in the `ssdp`, `ohmconnect`, `namecheapdns` and `ihc` components.

The denonavr manifest declares some ssdp values, but does not depend on ssdp. This could in fact be an upstream bug.

https://github.com/home-assistant/core/blob/dev/homeassistant/components/denonavr/manifest.json

**Comment by mweinelt** [comment]
I guess they expect `ssdp` to be loaded through `default_config`, which depends on it. Did you configure `default_config`? @colemickens

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/servers/home-assistant/component-packages.nix:323]: Fair enough. Thanks for clarification.

**Comment by Mic92** [comment]
There were two patch versions released in the meantime.

**Comment by mweinelt** [comment]
The bump has been pushed.

**Comment by colemickens** [comment]
Please don't consider me blocking. I was able to get `0.112.1` working after activating `ssdp`, so everything's good on my end. I'll pull `0.112.2` and test later, but I don't anticipate problems.

Thanks for the PR and the help!

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 91958` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
<br>- home-assistant
<br>- python37Packages.graphql-core
<br>- python37Packages.graphql-server-core
<br>- python37Packages.influxdb-client
<br>- python37Packages.rx
<br>- python38Packages.graphql-core
<br>- python38Packages.graphql-server-core
<br>- python38Packages.influxdb-client
<br>- python38Packages.rx
</details>

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 91958` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
<br>- home-assistant
<br>- python37Packages.graphql-core
<br>- python37Packages.graphql-server-core
<br>- python37Packages.influxdb-client
<br>- python37Packages.rx
<br>- python38Packages.graphql-core
<br>- python38Packages.graphql-server-core
<br>- python38Packages.influxdb-client
<br>- python38Packages.rx
</details>

**Pull request merged**